File: LeetCode/Java Solutions/Make The String Great/make.java
public class Solution {
    public String makeGood(String s) {
        StringBuilder stack = new StringBuilder();
        for (char c : s.toCharArray()) {
            int lastCharIndex = stack.length() - 1;
            // Check if the current and last character in the stack form a bad pair
            if (stack.length() > 0 && Math.abs(c - stack.charAt(lastCharIndex)) == 32) {
                stack.deleteCharAt(lastCharIndex); // Remove last character
            } else {
                stack.append(c);
            }
        }
        return stack.toString();
    }
}

File: LeetCode/Python Solutions/Make The String Great/make.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack = []
        for char in s:
            # Check if the current and last character in the stack form a bad pair
            if stack and abs(ord(char) - ord(stack[-1])) == 32:
                stack.pop()
            else:
                stack.append(char)
        return ''.join(stack)
